### PR TITLE
implement GameCreationInteractor

### PR DESCRIPTION
**Pull request by benxu3**
Implemented createNewGame and createSavedGame methods in GameCreationInteractor. Please refer to only the most recent commit by me for the relevant changes. [implemented createNewGame and createSavedGame in GameCreationInteractor](https://github.com/CSC207-2022F-UofT/course-project-the-best-monopoly-team/commit/60eda28243f0435448da985ba19dab42025a7851)

Made necessary changes to other files for required constructors, methods, etc... Added some TODOs.

Action Items: (can someone take over items 1 and 2? I will do item 3)

1. Implement/Refactor case 2 and case 3 of createSavedGame(). This will require someone with working knowledge of Tree structures to figure out how we will save treeData and optionData and load the relevant information. (ie. what index in the array corresponds to what information) Refer to comments I have added in saveGame under TextFileTranslator to get an understanding of what I mean. 
2. Implement CornerTiles class functionality for Go, Jail (inJail and justVisiting), goToJail, freeParking. I have created a simple base in CornerTiles to work off of.
3. I will add JavaDoc and Unit Tests for TextFileTranslator, GameCreationInteractor, and Property in follow-up PRs. 

Aside: Richard was responsible for the previous commits, which was essentially just refactoring of the code. As well, Susannah was responsible for making the base code of the GameCreationInteractor, but her branch was essentially aborted after too many merge errors.

### CODE REVIEW

File: src/main/java/Interactors/IOController.java [hunk 1]
```diff
@@ -11,10 +12,10 @@ public class IOController {
     private UseCaseInteractor interactor;
     private State currentState;
 
-    public IOController(){
+    public IOController(File file){
         this.output = "Welcome to Monopoly! Do you want to ";
         this.input = 0;
-        this.interactor = new UseCaseInteractor();
+        this.interactor = new UseCaseInteractor(new TextFileTranslator(file));
```
Comment (harshkpatel) [inline src/main/java/Interactors/IOController.java:18]: I am gonna assume that the use case interactor is the one that handles the cases for if a file should be read or not. I think for functionality wise this works for now, but I think a better choice for how the cases should be implemented is through the IO controller, which we can handle in a later PR.

**Review by harshkpatel — APPROVED** [review]
Overall looks good, besides the comment i left about how you are passing the files, which to repeat can be refactored with another PR (I am thinking that we just pass a null file if they want to create a new game), I will talk to Richard about that. 

All the other conventions look good and its good that you acknowledged the point about adding JavaDoc and Tests in a future PR. Great Work!

File: src/main/java/Interactors/GameCreation.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            Property cProperty = (Property) property;
+            for (int i=0; i <= 41; i++) {
+                if (cells.get(i) instanceof Property){
+                    Property currentCell = (Property) cells.get(i);
+                    if (currentCell.getName().equals(cProperty.getName()));
+                    cells.remove(i);
+                    cells.add(i, cProperty);
+                }
+            }
+        }
```
Comment (kRichard32) [inline src/main/java/Interactors/GameCreation.java:146]: What is the purpose of this for loop?

File: src/main/java/Interactors/GameCreation.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        for (Cell property : propertiesSoFar){
+            Property cProperty = (Property) property;
+            for (int i=0; i <= 41; i++) {
+                if (cells.get(i) instanceof Property){
+                    Property currentCell = (Property) cells.get(i);
+                    if (currentCell.getName().equals(cProperty.getName()));
+                    cells.remove(i);
+                    cells.add(i, cProperty);
+                }
+            }
```
Comment (kRichard32) [inline src/main/java/Interactors/GameCreation.java:145]: Should we separate this chunk into a different method? We might not need this for the case where we create a new game.

**Review by kRichard32 — APPROVED** [review]
Good uses of for loops in order to make the operation of loading more neat, and good organization of code in general. The organization allows for easy reading of the code, and the for loops are effective ways of handling replacing properties in the case of a loaded game. Other than the JavaDoc and commenting needed later, the code looks good to go!

**Review by derrcodes — APPROVED** [review]
Changes are good especially in terms of making everything seem more concise. It will help to maintain the product if required. I assume most changes are made for fixing merge conflicts and the modification of some instance attributes. In terms of function I think its all good but we still need to add documentation.

**Pull request merged**